### PR TITLE
Update from warp to axum

### PR DESCRIPTION
**Pull request by netsirius**
The crate [wrap](https://docs.rs/warp/latest/warp/) is no longer supported, so we are switching to using [axum](https://docs.rs/axum/latest/axum)

### CODE REVIEW

File: crates/locutus-core/src/client_events/websocket.rs [hunk 1]
```diff
@@ -42,13 +46,13 @@ impl WebSocketProxy {
     where
         T: Into<SocketAddr>,
     {
-        let filter = warp::filters::path::end().map(warp::reply::reply).boxed();
+        let filter = Router::default();
```
Comment (iduartgomez) [inline crates/locutus-core/src/client_events/websocket.rs:49]: Change mentions of filter to `route` everywhere.

File: crates/locutus-node/src/errors.rs [hunk 2]
```diff
@@ -1,34 +1,79 @@
-use warp::{hyper::StatusCode, reject::Reject, reply, Rejection, Reply};
+use axum::http::StatusCode;
+use axum::response::{Html, IntoResponse, Response};
+use locutus_stdlib::client_api::ErrorKind;
+use std::fmt::{Display, Formatter};
 
-use super::*;
+#[derive(Debug)]
+pub enum Error {
```
Comment (iduartgomez) [inline crates/locutus-node/src/errors.rs:7]: does this need o be public really? is enough with it being pub(crate) ?

File: crates/locutus-node/src/http_gateway.rs [hunk 3]
```diff
@@ -52,45 +56,54 @@ impl HttpGateway {
             response_channels: HashMap::new(),
         };
 
-        let get_home = warp::path::end().and_then(home);
-        let base_web_contract = warp::path::path("contract").and(warp::path::path("web"));
+        // let get_home = warp::path::end().and_then(home);
```
Comment (iduartgomez) [inline crates/locutus-node/src/http_gateway.rs:59]: remove all the comments if is no necessary anymore

File: crates/locutus-node/src/http_gateway.rs [hunk 4]
```diff
@@ -206,19 +242,11 @@ struct NewSubscription {
...
-        Ok(m) if m.is_binary() || m.is_text() => m.into_bytes(),
-        Ok(m) if m.is_close() => return Err(None),
-        Ok(m) if m.is_ping() => {
-            return Ok(Some(Message::pong(vec![0, 3, 2])));
-        }
-        Ok(m) => {
-            tracing::debug!(msg = ?m, "received random message");
-            return Ok(None);
-        }
+        Ok(m) => m.into_data(),
```
Comment (iduartgomez) [inline crates/locutus-node/src/http_gateway.rs:249]: we don't need to hndle pings manually anymore?

File: crates/locutus-node/src/errors.rs (hunk 2 above)
Comment (iduartgomez) [reply crates/locutus-node/src/errors.rs:7]: also we probably should use a more descriptive name for this, like `WebSocketApiError`

File: crates/locutus-core/src/client_events/websocket.rs (hunk 1 above)
Comment (netsirius) [reply crates/locutus-core/src/client_events/websocket.rs]: Name changed to router

File: crates/locutus-node/src/errors.rs (hunk 2 above)
Comment (netsirius) [reply crates/locutus-node/src/errors.rs]: The problem is that it is used outside of this crate

File: crates/locutus-node/src/http_gateway.rs (hunk 3 above)
Comment (netsirius) [reply crates/locutus-node/src/http_gateway.rs]: deleted

File: crates/locutus-node/src/http_gateway.rs (hunk 4 above)
Comment (netsirius) [reply crates/locutus-node/src/http_gateway.rs]: Fixed, it was incomplete

File: apps/stdlib/package.json [hunk 5]
```diff
@@ -12,7 +12,8 @@
     "bs58": "^5.0.0"
   },
   "devDependencies": {
-    "@types/jest": "^28.1.5",
+    "@types/jest": "^28.1.8",
+    "@types/mocha": "^10.0.1",
```
Comment (iduartgomez) [inline apps/stdlib/package.json:16]: I don't think we need for anything?

**Pull request merged**